### DISCUSSION TITLE
Request with a wrong content-type returns obscure error instead of 415 Unsupported Media Type

### DISCUSSION THREAD

**Discussion by notpushkin** [fastapi/fastapi#8586] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Works for any endpoint with JSON request body:

@app.post("/login")
def do_login(username: str, password: str):
    return {"token": "123"}
```

### Description

If you send an HTTP request with different content type(e. g. form data):
```shell
http --form POST 'http://127.0.0.1:8000/login' \
  'username=test' \
  'password=test'
```

You'll get a response:
```http
HTTP/1.1 422 Unprocessable Entity
content-length: 88
content-type: application/json
date: Fri, 11 Feb 2022 12:07:41 GMT
server: uvicorn

{
    "detail": [
        {
            "loc": [
                "body"
            ],
            "msg": "value is not a valid dict",
            "type": "type_error.dict"
        }
    ]
}
```

### Wanted Solution

Instead, it would be nice to return something along the lines of:

```http
HTTP/1.1 415 Unsupported Media Type
content-type: application/json

{
    "detail": "Unsupported media type: multipart/form-data (expected application/json)"
}
```

### Wanted Code

```python
# N/A
```

### Operating System

macOS

### FastAPI Version

0.70.1

### Python Version

Python 3.9.10

**Comment by jruizaranguren** [comment]
[There is a closed pull request ](https://github.com/tiangolo/fastapi/pull/4615) solving this issue that was closed without major explanation. It is weird that fastapi send 422 for requests using unsupported media-types.

What is the status of this question? Should be tackled, ignored?

**Reply by CorneeSean** [reply]
+1 for the question above ☝️
